src/lang/lang_values.ml

```diff
@@ -750,6 +754,17 @@ let rec check ?(print_toplevel = false) ~level ~(env : T.env) e =
                 x
         in
         e.t >: aux a.t
+    | Open (a, b) ->
+        check ~level ~env a;
+        a.t <: mk T.unit;
```

Comment:
I think it makes sense to restrict `open` to `unit` modules for now. We can change that if we ever see a need for it but it seems nice to say `unit` modules are "proper" modules, non-unit modules are decorated values.